src/app/core/signing.guard.ts

```diff
@@ -43,21 +43,24 @@ export class SigningGuard implements CanActivate {
       SigningGuard.key = linkStr.match(/key=([^&]*)/)[1];
     } catch (e) {
       // noinspection JSIgnoredPromiseFromCall
-      this.router.navigate([ '/invalid-link' ]);
+      this.router.navigate([ '/invalid-link' ]).catch(e => alert('Something went wrong, see the error below: \n' + e +
+        '\n please contact us at virtrolio.team@gmail.com if this problem persists.'));
     }
 
     /** Redirection based on authService.checkKey() & authService.isLoggedIn() */
     return this.authService.checkKey(SigningGuard.uid, SigningGuard.key).then(validKey => {
       if (validKey === false) {
         // noinspection JSIgnoredPromiseFromCall
-        this.router.navigate([ '/invalid-link' ]);
+        this.router.navigate([ '/invalid-link' ]).catch(e => alert('Something went wrong, see the error below: \n' + e +
+          '\n please contact us at virtrolio.team@gmail.com if this problem persists.'));
```

Comment:
Replace with standardized Error message
```suggestion
      this.router.navigate([ '/invalid-link' ]).catch(e => AuthService.displayError(e));
```